### ISSUE TITLE
The Achilles Heel

### ISSUE THREAD

**Issue by conartist6** [sithmel/iter-tools#15]
```
function iter(iterable) {
// ...
if (typeof iterable === 'object' && 'next' in iterable) {
  return iterable
}
```
Any object which has `next` as a key of itself or in its prototype chain will not work normally with `iter`. `next` is far too common a word to serve as a sentinel value, and `iter` is used all over the library.

I've just done a some tweaking which will result in a diff momentarily and it seems to me that there's no need to have this at all to get the test suite to pass. To remove it though would mean bumping the major version again.

**Comment by sithmel** [comment]
I am not sure I should use iter and async-iter everywhere. It was an old idea, but it doesn't looks so clever now. I need to double check what was the rationale behind it

**Comment by conartist6** [comment]
Well it's certainly convenient to be able to treat objects as iterables, though I don't think it's terribly inconvenient to be forced to call `entries` (or `keys` or `values`) on them.

Other than that it looks like it just clears up the difference between getting yourself an Iterator by calling a generator method vs. calling Symbol.iterator

**Comment by conartist6** [comment]
Closed by #16
